src/clj/rems/db/applications.clj

```diff
@@ -979,12 +979,14 @@
 (defn get-dynamic-application-state [application-id]
   (let [application (first (db/get-applications {:id application-id}))
         events (map fix-event-from-db (db/get-application-events {:application application-id}))
-        fixed-application (assoc application
-                                 :state ::dynamic/draft
-                                 :dynamic-events events
-                                 :workflow (fix-workflow-from-db (:workflow application)))]
-    (assert (is-dynamic-application? fixed-application))
-    (dynamic/apply-events fixed-application events)))
+        application (assoc application
+                           :state ::dynamic/draft
+                           :dynamic-events events
+                           :workflow (fix-workflow-from-db (:workflow application))
+                           :last-modified (or (:time (last events))
```

Comment:
How about taking the time of the last event with `:time`?

Reply:
It *is* taking the time of the last event with `:time`. What do you mean?

Reply:
It takes the last event, and then its time. Instead of taking the time of the last event with time.

Reply:
I don't understand. How would you write the code?

Reply:
`:last-modified (or (last (keep :time events))`

Reply:
Ok, so if the newest event doesn't have `:time`, then it will skip that and use the next newest event which has `:time`.

Reply:
All events should have a time, but we don't yet have validation for that, so we cannot be sure about that.

Reply:
That is why I suggest that you could take the time of the last event with time...